handler.go

```diff
@@ -274,6 +276,10 @@ func (h *Handler) ServeHTTP(responseWriter http.ResponseWriter, request *http.Re
 		receiver = interceptor.WrapStreamReceiver(ctx, receiver)
 	}
 	h.implementation(ctx, sender, receiver, clientVisibleError)
+	if sendHTTPInternalError {
+		// protocolHandler returned a nil sender, so we can't send a proper HTTP response.
+		responseWriter.WriteHeader(http.StatusInternalServerError)
```

Comment:
If the sender is nil, that means we wrap a no-op sender (which sends nothing across the wire). In that case, we should at least send back an error from the handler. Without this logic, we'd send back an empty HTTP 200 response.

I'm not sure if we want to encode more details about the `clientVisibleError` in a response body here but this should at least reflect that an error occurred on the client.

Reply:
It looks like the `clientVisibleError` could be a typed `*connect.Error`. If so, we should map its error code to the associated HTTP code rather than always using `http.StatusInternalError`. For example, I noticed that `clientVisibleError` is sometimes set to to the `timeoutErr` (which will often be initialized with `CodeInvalidArgument`).

However, the `clientVisibleError` is a detail of the `protocolHandler` that was used to serve the request (e.g. gRPC, Connect, etc). Fortunately, the gRPC and Connect [error codes](https://connect.build/docs/protocol/#error-codes) match, so we _should_ be able to use the same code transformation regardless of the `protocolHandler` that was used. I'd need to take a closer look at this though to refresh.

```go
responseWriter.WriteHeader(connectCodeToHTTP(CodeOf(clientVisibleError)))
```

However, this might be another place for concern if we ever expose another `protocolHandler` that doesn't have the same `connect.Code` -> HTTP Code transformation. We'll need to verify this behavior before we move forward here - I have a feeling we might not want to use the `responseWriter.WriteHeader` call in this context.